infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/receive/SinkWriter.java

```diff
@@ -29,12 +32,20 @@ public abstract class SinkWriter {
     // Registry table for verifying replicated entries from Source side.
     final CorfuTable<TableName, CorfuRecord<TableDescriptors, TableMetadata>> registryTable;
 
+    // Configuration for LR in Source / Sink cluster.
+    final LogReplicationConfig config;
+
     // Limit the initialization of this class only to its children classes.
-    SinkWriter(CorfuRuntime rt) {
+    SinkWriter(CorfuRuntime rt, LogReplicationConfig config) {
         this.rt = rt;
         // Instantiate TableRegistry and register serializers in advance to avoid opening
         // registry table in log entry writer's transaction
         this.registryTable = rt.getTableRegistry().getRegistryTable();
+        this.config = config;
+        if (config.getStreamsDiscoveryMode().equals(StreamsDiscoveryMode.DYNAMIC)) {
+            // In DYNAMIC mode signal the config to sync with registry table during initialization
+            config.syncWithRegistry();
```

Comment:
the sync will be done twice back-to-back, once each during the creation of LogEntrySync and SnapshotSync classes.  The config should be done only once and passed to both the classes.